app/sequencescape/sequencescape/api/v2/shared/has_requests.rb

```diff
@@ -57,18 +57,20 @@ def pool_id
     end
 
     # Finding 'in_progress' requests
```

Comment:
Think the existing comment here is actually a little misleading. It's not selecting based on state, it's selecting requests of a particular request type.

Nice refactor though 👍

Reply:
Credit to Code pilot, it is good at refactoring. Much clearer if a few more lines.

Reply:
Will change the comment yes.